GSA-TTS/FAC
# Allowing users to delete audit editors

Draft: Coordinating merge with static site update

Addresses https://github.com/GSA-TTS/FAC/issues/3409

In this PR:
- Adding a link in the audit access management page that lets an editor remove another editor, as long as it's not yourself
- Fixed centering for the Edit column
- Unit tests

Testing:
- Go to the user management page for an audit you've created (this way you are an editor.) Removal links should only exist for those with the editor role.
- Click the Remove link for your user's row and you should get a page saying you're not allowed to. You can use devtools to un-disable the button and submit, but it will error. Now is also a good time to confirm that clicking the cancel button returns you to the user management page.
- Add a new auditor with some test email. Clicking Remove for that editor should show the normal removal page. Clicking the submit button should remove them and return you to the use management page. You can see that they've been added to the`audit_deletedaccess` table as well.
- Trying to get to a removal page for an editor that doesn't exist for this audit should 404. To do this, change the `id` param to some random number. This should also happen if you use the `id` for a non-editor user, which you can find in `audit_access`.
- If you're logged in as someone with audit access who is NOT an editor (i.e. a certifier), you shouldn't see the remove column/links at all. If you try to go to a removal url directly, it will error. I think the easiest way to test this is by having the Cypress test create an audit so it can make your normal login email a certifier instead of an editor.

## PR Checklist: Submitter


## PR Checklist: Reviewer


The larger the PR, the stricter we should be about these points.

## Pre Merge Checklist: Merger
